src/azure-cli/azure/cli/command_modules/aro/_rbac.py

```diff
@@ -29,17 +34,63 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id, label):
         name=CONTRIBUTOR,
     )
 
-    for assignment in list(client.role_assignments.list_for_scope(vnet)):
-        if assignment.role_definition_id.lower() == role_definition_id.lower() and \
-                assignment.principal_id.lower() == object_id.lower():
-            return
+    if has_assignment(auth_client.role_assignments.list_for_scope(vnet), role_definition_id, object_id):
+        return
 
-    # use object_id as a namespace and vnet and label as a salt for generating UUID to
-    # make it stable across calls
-    role_uuid = uuid.uuid5(uuid.UUID(object_id), ','.join(vnet + label))
+    # generate random uuid for role assignment
+    role_uuid = _gen_uuid()
```

Comment:
Wasn't sure if we should keep the previous method of generating a role_uuid.
```
role_uuid = uuid.uuid5(uuid.UUID(object_id), ','.join(vnet + label))
```

Reply:
no - all references to uuid5 should be removed.

Reply:
Yes use the _gen_uuid() is it later used in test recording as it can be mocked.

Reply:
@petrkotas Thanks!  I've added the mock and rerunning with `--live` to rerecord.  Will update when done.